### PR TITLE
Generate Proxy Headers lookup form/version Id when provided submissionId

### PR DESCRIPTION
**Pull request by usingtechnology**
# Description
Error when submitting from a draft on a form that has an External API - seeing 400 errors.

The issue is loading that page has no form id context, thus the form id is not passed into the generate proxy headers function. When parsing the generated proxy headers (to resolve which external api to call) the external api cannot be found because there is no form id!

The fix is: when provided a submissionId, go find the form and version id and use those values in the generated proxy headers.

## Types of changes

fix (a bug fix)

## Checklist

- [x] I have read the [CONTRIBUTING](/bcgov/common-hosted-form-service/blob/main/CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have run the npm script lint on the frontend and backend
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have approval from the product owner for the contribution in this pull request

## Further comments

JIRA: https://bcdevex.atlassian.net/browse/FORMS-1445?atlOrigin=eyJpIjoiYWE5ODM2NTE3MjNjNGY0NzgzMmI0NzAwZjA5ZmQxYzIiLCJwIjoiaiJ9

Fider: https://chefs-fider.apps.silver.devops.gov.bc.ca/posts/177/external-api-not-working-in-drafts

CCP Pages wiki: https://bcdevex.atlassian.net/wiki/spaces/CCP/pages/1553924326/External+API+does+not+work+in+Drafts

### CODE REVIEW

File: app/src/forms/proxy/service.js [hunk 1]
```diff
@@ -19,15 +19,38 @@ const trimLeadingSlashes = (str) => str.replace(/^\/+|\$/g, '');
...
+    // when we are provided with a submission id (ex. submitting a draft submission)
+    // we need to fetch the related form and version id (if not provided)
+    if (submissionId) {
+      const meta = await SubmissionMetadata.query().where('submissionId', submissionId).first();
+      if (meta) {
+        formId = meta.submissionId;
+        versionId = meta.formVersionId;
+        submissionId = meta.submissionId;
+      }
+    }
```
Comment (WalterMoar) [inline app/src/forms/proxy/service.js:35]: The code coverage report is complaining about these lines:

image

**Review by WalterMoar — CHANGES_REQUESTED** [review]
Makes sense to me, but would love to see the new code covered by the tests.

File: app/src/forms/proxy/service.js (hunk 1 above)
Comment (usingtechnology) [reply app/src/forms/proxy/service.js:35]: how do i see the code coverage report?

Comment (WalterMoar) [reply app/src/forms/proxy/service.js:35]: I'll create a Task to add that to the docs!
- locally for the backend it's in `/app/coverage/lcov-report/index.html`
- for a PR it's in the artifacts for the "Tests" Action:
image

Comment (usingtechnology) [reply app/src/forms/proxy/service.js]: probably on the long list of backlog items but maybe the metrics should somehow be in your face? like easily seeing that this PR reduced coverage by X% or something?

**Comment by usingtechnology** [comment]
> Makes sense to me, but would love to see the new code covered by the tests.

added tests and good thing... showed a bug to fix. :)

File: app/src/forms/proxy/service.js (hunk 1 above)
Comment (WalterMoar) [reply app/src/forms/proxy/service.js]: That would be a great feature! A lot of stuff happens after the merge, etc, due to permissions. But yes, it's something that would be great if someone looked into it.

**Review by WalterMoar — APPROVED** [review]
Looks good! I'm not sure how to try it out but assume that kind of testing has been done.

**Comment by usingtechnology** [comment]
brian and i are testing manually. awaiting his result (mine was good) before merging.

**Comment by usingtechnology** [comment]
works for Brian, so will be merging.

**Pull request merged**